packages/nifti-volume-loader/src/helpers/scaleNiftiArray.ts

```diff
@@ -0,0 +1,25 @@
+/**
+ * Given a pixel array, rescale the pixel values using the rescale slope and
+ * intercept
+ *
+ * Todo: add the scaling of PT and SUV
+ * @param array - The array to be scaled.
+ * @param scalingParameters - The scaling parameters
+ * @returns The array being scaled
+ */
+export default function scaleNiftiArray(
+  array: Float32Array | Int16Array | Uint8Array,
+  niftiHeader
+): void {
+  const arrayLength = array.length;
+  const { scl_slope, scl_inter } = niftiHeader;
+
+  if (scl_slope === 0 || Number.isNaN(scl_slope)) {
+    // No scaling encoded in NIFTI header
+    return;
+  }
+
+  for (let i = 0; i < arrayLength; i++) {
```

Comment:
You are modifying this in place - would it be safer to make a copy?  What if the data is exposed as a Uint8array and the scaling no longer works?  This just feels like it might cause problems, but the memory usage is definitely bigger if you copy the data instead.

Reply:
This is the same behaviour in the dicom image loader